frontend/public/components/ThemeProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+
+export const THEME_USER_SETTING_KEY = 'console.theme';
+export const THEME_LOCAL_STORAGE_KEY = 'bridge/theme';
+const THEME_SYSTEM_DEFAULT = 'systemDefault';
+const THEME_DARK_CLASS = 'pf-theme-dark';
+const THEME_DARK = 'dark';
+
+export const updateThemeClass = (htmlTagElement: HTMLElement, theme: string) => {
+  let systemTheme: string;
+  if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+    systemTheme = THEME_DARK;
+  }
+  if (theme === THEME_DARK || (theme === THEME_SYSTEM_DEFAULT && systemTheme === THEME_DARK)) {
+    htmlTagElement.classList.add(THEME_DARK_CLASS);
+  } else {
+    htmlTagElement.classList.remove(THEME_DARK_CLASS);
+  }
+};
+
+export const ThemeContext = React.createContext<string>('');
+
+export const ThemeProvider = ({ children }) => {
+  const htmlTagElement = document.documentElement;
+  const localTheme = localStorage.getItem(THEME_LOCAL_STORAGE_KEY);
+  const [theme, , themeLoaded] = useUserSettings(
+    THEME_USER_SETTING_KEY,
+    THEME_SYSTEM_DEFAULT,
+    true,
+  );
+  const mqListener = React.useCallback(
+    (e) => {
+      if (e.matches) {
+        htmlTagElement?.classList.add(THEME_DARK_CLASS);
+      } else {
+        htmlTagElement?.classList.add(THEME_DARK_CLASS);
```

Comment:
When switching from dark to light in my system it doesn't work. I expect this copy and paste issue is the reason. Maybe we can reuse updateThemeClass here?

```suggestion
        htmlTagElement?.classList.remove(THEME_DARK_CLASS);
```